client/src/plugins/error-tracking/ErrorTracking.js

```diff
@@ -186,7 +195,39 @@ export default class ErrorTracking extends PureComponent {
     return this.recheckSentry();
   }
 
+  normalizeEventPath = (event) => {
+    try {
+
+      const { exception, request } = event;
+      const { values } = exception;
+
+      request.url = normalizeUrl(request.url);
+
+      values.forEach((exceptionVal) => {
+        const { stacktrace } = exceptionVal;
+        const { frames } = stacktrace;
+
+        frames.forEach((frame) => {
+          frame.filename = normalizeUrl(frame.filename);
+        });
+      });
+
+      return event;
+    } catch (err) {
+
+      // Don't loose the actual event in case things go wrong
+      return event;
```

Comment:
Is it intended to fail? Otherwise, this will hide errors from this implementation we might not see in the end.

Reply:
I don't see a reason why the normalization fails (it's just a regex replace operation). However if for some reason it fails, we still want to send the non-normalized original error event to Sentry. In that case we'd see a "Source code not found" error in Sentry and check why this operation failed using the actual data.

Reply:
meta-question: When this piece of code would throw an error, because the normalization failed, would it then send this error to sentry?

Reply:
Original event that we're trying to normalize: Yes
The error that we caused while normalizing the error: No

Rational of No:
Since this method is executed before we send events to Sentry, if we send additional stuff to Sentry at this point, we might easily cause some sort of infinite loop. IMO it's best if we don't send and additional error here. If we detect we're unable to normalize data through Sentry (the path we see would not be normalized then), we can investigate why this happened using the actual data itself. That's not very likely TBH. I was just being extra-cautious there.

Reply:
Does it mean that if for some reason the block above fails, we send full path which may contain personal data?

Reply:
Yes, we send the event as is in that case. I'm not sure if this is considered personal data or not. It's just the application path.

Reply:
I mean if it's like:

`/Users/maciej-barelkowski/workspace/bpmn-io/camunda-modeler`

then it's not anonymous anymore.

Reply:
Cf. https://sentry.io/organizations/camunda-modeling/issues/1694263988/?project=5223041&query=is%3Aunresolved

Reply:
Why not simply raise an error if the before-event fails. Our test cases should be good enough to detect this before production.

Reply:
To be honest, I would just make sure that the try block never fails. Or maybe it already doesn't?

Reply:
![infinite](https://user-images.githubusercontent.com/15003836/85716994-518e3500-b6ed-11ea-8d34-f0379e5e96c5.png)

This is why I wanted to make sure we always catch errors here (not that it'll ever fail). In case the normalization fails, we might end up having some sort of infinite loop. Any opinions?

Reply:
Here's the solution we discussed: https://github.com/camunda/camunda-modeler/pull/1846/commits/5460106b3f3e38c86adb45cb9a7bf79851c3eea9

Reply:
Good job!